lexicon/__main__.py

```diff
@@ -36,13 +38,10 @@ def BaseProviderParser():
     return parser
 
 def MainParser():
-    current_filepath = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'providers')
-    providers = [os.path.splitext(f)[0] for f in os.listdir(current_filepath) if os.path.isfile(os.path.join(current_filepath, f))]
-    providers = list(set(providers))
-    providers.remove('base')
-    providers.remove('__init__')
-    providers = [x for x in providers if not x.startswith('.')]
-
+    providers = []
+    for _, modname, _ in pkgutil.iter_modules(providers_package.__path__):
```

Comment:
this is neat, I haven't seen this before. I'm assuming its supported by python 2.x and 3.x?

Reply:
Yes, it is working on both. I developed using both py27 on Debian and py37 on Windows. And the CI is testing both versions anyway.